### ISSUE TITLE
Screen reader does not announce the status message indicating the number of results:A11y_Aspire Dashboard_Resources_Screenreader.

### ISSUE THREAD

**Issue by AnKushSingh05** [dotnet/aspire#10142]
### GitHubTags
#A11ySev2; #A11yTCS;#DesktopWeb; #Win32;#.NET Project system for CSharp and VB;#BM_AspireDashboard_Web_June25;#Win32;#WCAG4.1.3;#ScreenReader;#NVDA;#Narrator;#StatusMessage;#A11yMAS;

### Actual Result:
Screen reader does not announce the status message indicating the number of results

### Expected Result:
The screen reader should correctly announce the status message as for example: 2 results found.

### Environment Details:
Application Name: Aspire Dashboard
Edition Windows 11 Enterprise
Version 24H2
OS build 26100.4351

### Repro Steps:

1. Open Visual studio.
2. Open a aspire project and run it.
3. A new window will open in the chromium edge, TAB to "Resources" tab item and hit ENTER key.
4. Verify that Screen reader does not announce the status message indicating the number of results

### User Impact:
Visually impaired users who rely on screen readers will not the results information when we search in the filter.

### Attachment:

https://github.com/user-attachments/assets/4211056d-77c3-4e0e-a83c-f2863eaf203c
